File: java/eligibility.java
import java.util.Scanner;

public class eligibility {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int cases = scan.nextInt();
		while (cases-- > 0) {
			String name = scan.next();
			String postDate = scan.next();
			String DoB = scan.next();
			int courses = scan.nextInt();
			int postYear = Integer.parseInt(postDate.split("/")[0]);
			int DoBYear = Integer.parseInt(DoB.split("/")[0]);
			if (postYear >= 2010 || DoBYear >= 1991)
				System.out.printf("%s eligible%n", name);
			else if (courses >= 41)
				System.out.printf("%s ineligible%n", name);
			else
				System.out.printf("%s coach petitions%n", name);
		}
		scan.close();
	}
}

File: ts/src/eligibility.ts
import readline = require('readline')

let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      const [name, date, dob, c] = line.split(' ')
      const courses = parseInt(c ?? '0')
      const postYear = parseInt(date?.split('/')[0] ?? '0')
      const dobYear = parseInt(dob?.split('/')[0] ?? '0')

      if (postYear >= 2010 || dobYear >= 1991) {
        console.log('%s eligible', name)
      } else if (courses >= 41) {
        console.log('%s ineligible', name)
      } else {
        console.log('%s coach petitions', name)
      }
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
